### ISSUE TITLE
Expose boolean of whether an item is currently visible in the grid

### ISSUE THREAD

**Issue by sindresorhus** [spacenation/swiftui-grid#77]
I have a large grid of `NSImageView`'s that play animated GIFs. This is very resource intensive, so I would like to pause playing the GIF animation for items that are outside the viewport. It would be useful if Grid could somehow expose this information.

**Comment by sindresorhus** [comment]
It would also be useful with a method that runs a callback when an element appears in the viewport, since `.onAppear()` triggers for all grid elements at load time and not when they enter the viewport.

**Comment by victormihaita** [comment]
I try to paginate my API calls using this logic and is impossible because I cannot know when the last item in my dataSource is displayed. I think this is an important fix as well as #81

**Comment by ay42** [comment]
@victormihaita most likely this feature will not be implemented. Just like HStack and VStack, Grid is a lower level layout element. I imagine this can be done with the ViewModifier on the presented views directly but not as a part of this project.

**Comment by ay42** [comment]
@victormihaita I'm looking for a way to fix #81, do you know how to do it?

**Comment by victormihaita** [comment]
In this code, when calling onAppear, I expect that it should be called only for the elements visible on the screen, but instead is called for all the elements in the grid. is that correct?

```
Grid(viewModel.dataSource, id: \.id) {
    CardView($0).onAppear {
        //Do something
    }
}
```
